Adds tier policy document (#341)
algebraic-solving/msolve: doc/tier-policy.md

# Tier policy

## Tiers

msolve provides three tiers of target support:

- msolve provides no guarantees about tier 3 targets; they exist in the codebase, but may or may not build.
- msolve's continuous integration checks that tier 2 targets will always build, but they may or may not pass tests.
- msolve's continuous integration checks that tier 1 targets will always build and pass tests.

Adding a new tier 3 target imposes minimal requirements; we focus primarily on avoiding disruption to other ongoing msolve development.

Tier 2 and tier 1 targets place work on msolve project developers as a whole, to avoid breaking the target. Thus, these tiers require commensurate and ongoing efforts from the maintainers of the target, to demonstrate value and to minimize any disruptions to ongoing msolve development.

This policy defines the requirements for accepting a proposed target at a given level of support.

## Software platforms

### Tier 1
- x86_64 Linux

> - x86_64 macOS

Reviewer: FYI x86_64 macOS has been downgraded to Tier 2 platform for Rust in 2025 (see https://blog.rust-lang.org/2025/08/19/demoting-x86-64-apple-darwin-to-tier-2-with-host-tools/).  Will we have to do the same soon?

Author: Well, currently x86_64 is supported by macOS, but once it is no longer officially supported, yes, it has to become tier 2.

- ARM64 macOS

### Tier 2
- i686 Linux

> - x86_64 FreeBSD

Reviewer: Has msolve ever been tested on FreeBSD? I can't find it in FreshPorts.

- ARM64 FreeBSD

> - PPC64 Linux

Reviewer: The big-endian one? It seems to have become pretty obsolete...

Reviewer: Indeed, Julia only supports the little-endian ppc64le. But this should no longer matter after the removal of Tier 2, anyway.

> - PPC64 Linux

Reviewer: ```suggestion
- PPC64le Linux
```

Reviewer: Also, ARM64 Linux should at least be in Tier 2.

> - RISC64 Linux

Reviewer: ```suggestion
- RISCV64 Linux
```

It is also surprising to see RISC-V in Tier 2. Do we happen to have a board, and if so, what extensions does it have?

Reviewer: Also, given that ARM64 Linux is much more widespread, would it make sense to classify it under Tier 1/2 as well?

Author: Typo is fixed.

# Tier policy

## Tiers

msolve provides three tiers of target support:

- msolve provides no guarantees about tier 3 targets; they exist in the codebase, but may or may not build.

> - msolve's continuous integration checks that tier 2 targets will always build, but they may or may not pass tests.

Reviewer: ```suggestion
- msolve expects that tier 2 targets will always build, but they may or may not pass tests.
```

I would soften this statement as Yggdrasil only performs post-release CI checks (and it's not msolve's CI anyway). Otherwise, LGTM.
